src/test/clojure/clara/test_accumulators.clj

```diff
@@ -284,3 +284,48 @@
     (is (= 1 (count temp-history)))
     (is (= [{:?his (->TemperatureHistory [temp-10-mci])}]
             temp-history))))
+
+(deftest test-grouping-accum
+  (let [grouping-accum (acc/grouping-by :temperature)
+        grouping-query (dsl/parse-query [] [[?t <- grouping-accum
+                                             :from [Temperature]]])
+
+        convert-return-fn (fn [m]
+                            (if (seq m)
```

Comment:
Looks like either the guard or a default value is needed here since max-key requires is not defined for zero arguments. We could change the structure a bit to make that more obvious (one option is shown in the following snippet), but I don't think it's a big difference.

```clj
        convert-return-fn (fn [m]
                            (if (empty? m)
                              ;; No max temperature to convert, so return an empty sequence.
                              []
                              (val (apply max-key key m))))
```